### PR TITLE
Config and logging improvements

### PR DESCRIPTION
**Pull request by esmet**
- Colorized logging
- Context-based config management that closely mirrors kubectl

### CODE REVIEW

File: ankh/cli.py [hunk 1]
```diff
@@ -117,46 +178,138 @@ def template_ingress_hosts(global_config):
...
+            return r
+    return 0
+
+def run(base_dir, args):
+    logger.info("Running from directory %s" % os.getcwd())
+
+    logger.info("Gathering local configuration...")
+    global_config = gather_config(args, log=True)
+    if 'dependencies' in global_config:
+        logger.error("Base directory %s contains dependencies in its config, but we're processing dependencies right now. Recursive dependencies aren't supported, so eliminate them before proceeding." % os.getcwd())
```
Comment (chrisdopuch) [inline ankh/cli.py]: Might mention something about being in the global config to make the error messaging a little clearer for users

File: ankh/cli.py [hunk 2]
```diff
@@ -24,36 +26,93 @@
...
 logger = logging.getLogger('ankh')
+valid_commands = [ 'apply', 'template', 'config', 'current-context', 'set-context' ]
+default_config_files = [ 'ankh.yaml', 'deploy.yaml' ]
+
+def ankh_config_read(args, log=False):
+    # Kube context may be present in this user's ankh config. Check for it.
+    if os.path.exists(args.ankhconfig):
+        with open(args.ankhconfig, 'r') as f:
+            config = yaml.safe_load(f)
+            if 'current-context' in config:
```
Comment (jondlm) [inline ankh/cli.py]: Shouldn't we just expect `current_context` to be in the config yaml directly?

File: ankh/cli.py [hunk 3]
```diff
@@ -201,14 +354,18 @@ def main():
...
-        if args.command == 'template':
+        if command == 'template':
             return template_command(global_config, args)
     except KeyboardInterrupt:
         logger.info("Interrupted")
         sys.exit(1)
 
-    logger.error("Unknown command %s" % args.command)
+    logger.error("Unknown command: '%s'" % command)
+    suggestion = difflib.get_close_matches(command, valid_commands, n=1)
```
Comment (jondlm) [inline ankh/cli.py]: Nice, even added a suggestion mechanism

File: scripts/kapplied [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
```
Comment (jondlm) [inline scripts/kapplied]: Cool, a script to grab the last applied configuration. I like it

File: ankh/cli.py (hunk 3 above)
Comment (esmet) [reply ankh/cli.py]: this is what happens at 2am

File: scripts/kapplied (hunk 4 above)
Comment (esmet) [reply scripts/kapplied]: at some point soon I'd like to compose this with `ankh template` to extract a single object, convert it to JSON, and then diff agains the last applied configuration.

File: ankh/cli.py (hunk 1 above)
Comment (esmet) [reply ankh/cli.py]: good call.

Comment (esmet) [reply ankh/cli.py]: done

File: ankh/cli.py (hunk 2 above)
Comment (esmet) [reply ankh/cli.py]: are you saying we should expect current-context as a top-level key in ~/.ankh/config (or $ANKHCONFIG / --ankhconfig)? that should be the case, but we're also resilient to that key being missing. you can add that key manually or use set-context which will write the key in for you. is that what you meant?

Comment (jondlm) [reply ankh/cli.py]: Sorry, I wasn't very clear on my comment. I just meant it should have an underscore instead of a dash

Comment (esmet) [reply ankh/cli.py]: kubectl's ~/.kube/config uses 'current-context' so I went with that

**Pull request merged**